src/App.js

```diff
@@ -5,19 +5,35 @@ import ChatLog from './components/ChatLog';
 
 const App = () => {
   const [totalLikes, setTotalLikes] = useState(0);
+  const [messages, setMessages] = useState(chatMessages); 
 
-  const handleLikeCount = (id, liked) => {
-    setTotalLikes((prevLikes) => (liked ? prevLikes + 1 : prevLikes - 1));
+  const handleLikeCount = (id) => {
+    const copyMessages = [...messages];
+    let newLiked;
+    for (let i = 0; i < copyMessages.length; i++){
+      if (copyMessages[i].id === id){
+        const oneMessage = {...copyMessages[i]}
+        oneMessage.liked = ! oneMessage.liked; 
+        copyMessages[i] = oneMessage;
+        newLiked = oneMessage.liked;
+      }
+    }
+
+    
+    setMessages(copyMessages); 
+    setTotalLikes((prevLikes) => (newLiked ? prevLikes + 1 : prevLikes - 1));
```

Comment:
Note that we can calculate the total number of liked counts from the current state of the list of messages on each render, rather than managing a separate piece of state for it. It may seem more efficient to track a piece of state for the count, rather than needing to iterate over the list of messages to calculate it on every render, but keep in mind that React will need to iterate over the entire messages array to generate our list of ChatEntry components anyway, so this effectively adds only a small additional sum to work we already need to do. Remember that from a complexity standpoint, it makes no difference whether we iterate once and perform two operations in the loop, or iterate twice and perform a single operation (so looping and summing, then later looping and building ChatEntry components have the same cost, and the summing is negligible). It's true that if there were other state changes that could occur and this forced our app to re-render the entire message list each time, we could start to see performance problems, but React provides ways to help alleviate this. So as a starting point, it's generally best to calculate any values that we can from the fewest possible pieces of state.

In order to do so, we could write a helper function to iterate through the message data and count the number of liked messages, or we can leverage built-in `reduce` method to help as follows.

```js
  // this code can be placed in the main body of the component, anywhere before the return
  // it creates a local variable that holds the calculated count of liked messages
  // adding a bool to a number treats true as 1 and 0 as false
  const totalLikes = messages.reduce((totalLikes, message) => (totalLikes + message.liked), 0);
```